app/src/features/settings/components/BillingTeam/components/BillingDetails/index.tsx

```diff
@@ -0,0 +1,102 @@
+import { useCallback, useEffect, useMemo } from "react";
+import { useSelector } from "react-redux";
+import { getUserAuthDetails } from "store/selectors";
+import { useLocation, useParams, useSearchParams } from "react-router-dom";
+import useBillingTeamFetcher from "../../hooks/useBillingTeamFetcher";
+import { getAvailableBillingTeams, getBillingTeamMemberById } from "store/features/billing/selectors";
+import { Result } from "antd";
+import { MyBillingTeamDetails } from "./MyBillingTeamDetails";
+import { OtherBillingTeamDetails } from "./OtherBillingTeamDetails";
+import APP_CONSTANTS from "config/constants";
+import { toast } from "utils/Toast";
+import { trackJoinBillingTeamRequestToastViewed } from "features/settings/analytics";
+import { getFunctions, httpsCallable } from "firebase/functions";
+import Logger from "lib/logger";
+
+const BILLING_TEAM_JOIN_REQUEST_ACTION = {
+  ACCEPT: "accept",
+  REJECT: "reject",
+};
+
+export const BillingDetails = () => {
+  const location = useLocation();
+  const { billingId } = useParams();
+  const user = useSelector(getUserAuthDetails);
+  const billingTeams = useSelector(getAvailableBillingTeams);
+  const [queryParams] = useSearchParams();
+
+  const joiningRequestAction = queryParams.get("joinRequestAction");
+  const userId = queryParams.get("userId");
+
+  const showReviewResultToast = useCallback((message: string, type: string, duration = 5) => {
+    switch (type) {
+      case "success":
+        toast.success(message, duration);
+        break;
+      case "warning":
+        toast.warn(message, duration);
+        break;
+      case "error":
+        toast.error(message, duration);
+        break;
+      default:
+        toast.info(message, duration);
+    }
+  }, []);
+
+  const hasAccessToBillingTeam = useMemo(
+    () =>
+      billingTeams?.some((billingTeam) => {
+        return billingTeam?.id === billingId && location.pathname !== APP_CONSTANTS.PATHS.SETTINGS.BILLING.RELATIVE;
+      }),
+    [billingTeams, billingId, location.pathname]
+  );
+  const isTeamMember = useBillingTeamFetcher(hasAccessToBillingTeam, billingId);
```

Comment:
Instead of separate checks, `useBillingTeamFetcher` can be modified for both the usecases. Please rename the hook appropriately